data/data/aws/bootstrap/main.tf

```diff
@@ -11,6 +11,24 @@ locals {
   volume_type      = "gp3"
   volume_size      = 30
   volume_iops      = local.volume_type == "io1" ? 100 : 0
+
+  // s3 object supports only 10 tags. The first 9 tags from 
+  // the list of aws_extra_tags are used for s3 object
+  new_tag_len = 9
+
+  keys  = flatten([
+      for key, value in local.tags: "${key}"
```

Comment:
To preserve the owned tag, we must get the slice of 8 user tags and then add in the owned tag.  If we add the owned tag to the user tags and then take the first 9, the owned tag may not be within the first 9 (based on the lexicographical ordering).

Reply:
True. Thanks for the check here. Updated the sliced map of tags creation.